src/libs/ReportListUtils.js

```diff
@@ -30,63 +47,149 @@ function getParticipantLogins(report) {
 }
 
 /**
- * Does the report have multiple participants?
+ * Returns the personal details for an array of logins
  *
- * @param {Object} report
- * @returns {Boolean}
+ * @param {Array} logins
+ * @returns {Array}
  */
-function reportHasMultipleParticipants(report) {
-    return getParticipantLogins(report).length > 1;
+function getPersonalDetailsForLogins(logins) {
+    return _.map(logins, (login) => {
+        let personalDetail = personalDetails[login];
+
+        if (!personalDetail) {
+            personalDetail = {
+                login,
+                displayName: login,
+                avatarURL: getDefaultAvatar(login),
+            };
+        }
+
+        return personalDetail;
+    });
 }
 
 /**
- * Is the login provided a report participant?
+ * Returns a string with all relevant search terms
  *
  * @param {Object} report
- * @param {String} login
- * @returns {Boolean}
+ * @param {Array} personalDetailList
+ * @return {String}
  */
-function isUserReportParticipant(report, login) {
-    return _.some(getParticipantLogins(report), participantLogin => participantLogin === login);
-}
+function getSearchText(report, personalDetailList) {
+    const searchTerms = [];
 
-/**
- * Check if the report has a draft comment
- *
- * @param {Object} comments
- * @param {Number} reportID
- * @returns {Boolean}
- */
-function hasComment(comments, reportID) {
-    const allComments = lodashGet(comments, `${ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT}${reportID}`, '');
-    return allComments.length > 0;
+    _.each(personalDetailList, (personalDetail) => {
+        searchTerms.push(personalDetail.displayName);
+        searchTerms.push(personalDetail.login);
+    });
+
+    if (report) {
+        searchTerms.push(...report.reportName.split(',').map(name => name.trim()));
+        searchTerms.push(...report.participants);
+    }
+
+    return _.unique(searchTerms).join(' ');
 }
 
 /**
  * Creates a report list option
  *
  * @param {Array<Object>} personalDetailList
  * @param {Object} [report]
- * @param {Object} [comments]
- * @param {Number} [activeReportID]
  * @returns {Object}
  */
-function createOption(personalDetailList, report, comments, activeReportID) {
-    // There is only one participant so return the single participant view
+function createOption(personalDetailList, report) {
+    const hasMultipleParticipants = personalDetailList.length > 1;
     const personalDetail = personalDetailList[0];
-    const isSingleUserDM = personalDetailList.length < 2;
     return {
         text: report ? report.reportName : personalDetail.displayName,
-        alternateText: (!report || isSingleUserDM) ? personalDetail.login : report.reportName,
+        alternateText: (!report || !hasMultipleParticipants) ? personalDetail.login : report.reportName,
```

Comment:
If no report or not having multiple participants (so singleUserDM). Hmm confusing to read. Also we know that report.login should be the same as personalDetail.login for singleUserDM. If we go with the single variable suggestion this can be made easier to read like so,
```
hasMultipleParticipants ? reportOrPersonalDetail. reportName : reportOrPersonalDetail.login;
```

Reply:
Yeah I think maybe they should be separated into two methods and it will be clearer.

Reply:
Yeah maybe, anyway i think its fine for now because it seems like there is a chance this could change soon as well.